std/src/main/scala/algebra/std/map.scala

```diff
@@ -26,33 +50,80 @@ class MapEq[K, V](implicit ev: Eq[V], a: AdditiveMonoid[V]) extends Eq[Map[K, V]
     check(x, y) && check(y, x)
```

Comment:
So... @avibryant and @tixxit both suggested that we default to "strict" equality, meaning that `Map()` and `Map("foo" -> 0)` would not be equal.

What do you think?